internal/ignition/ignition.go

```diff
@@ -615,8 +637,19 @@ func (g *installerGenerator) updateBootstrap(bootstrapPath string) error {
 			}
 
 			// get corresponding host
-			var host *models.Host
-			if bmhIsMaster(bmh) {
+			var (
+				host                             *models.Host
+				masterHostnames, workerHostnames []string
+			)
+			masterHostnames, err = getHostnames(masters)
+			if err == nil {
+				workerHostnames, err = getHostnames(workers)
+			}
+
+			if err != nil {
+				return err
+			}
+			if bmhIsMaster(bmh, masterHostnames, workerHostnames) {
```

Comment:
it is not clear why we need both masterHostnames & workerHostnames in order to check if bmhIsMaster and not checking it only in masterHostnames - please add a comment

Reply:
Since there is backward compatibility logic afterwards, we need to verify that the host is not a worker before falling back to the backward compatibility logic.

Reply:
There is a comment before the backward compatibility logic that explains that